File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/injection/modules/PreferencesModule.java
package com.crpr.androidcinema.injection.modules;

import android.content.Context;

import com.crpr.androidcinema.BuildConfig;
import com.crpr.androidcinema.data.preferences.PreferencesService;
import com.google.gson.Gson;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by cribeiro on 05/01/2016.
 */
@Module
public class PreferencesModule {

    @Provides
    @Singleton
    PreferencesService providePreferencesService(Context context, Gson gson){
        return new PreferencesService(context, gson, BuildConfig.APPLICATION_ID);
    }

}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/injection/modules/PreferencesModule.kt
package com.crpr.androidcinema.injection.modules

import android.content.Context
import com.crpr.androidcinema.BuildConfig
import com.crpr.androidcinema.data.preferences.AppPreferences
import com.crpr.androidcinema.data.preferences.PreferencesService
import com.google.gson.Gson
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by cribeiro on 05/01/2016.
 */
@Module
class PreferencesModule {

    @Provides
    @Singleton
    internal fun providePreferencesService(context: Context, gson: Gson): PreferencesService {
        return AppPreferences(context, gson, BuildConfig.APPLICATION_ID)
    }

}
